### PR TITLE
resolved name vs. id issue for family.js

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes partially #137 (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/row.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ class Row {
     options = options || {};
 
     families.forEach(function(family) {
-      const familyData = (data[family.name] = {});
+      const familyData = (data[family.id] = {});
```
Comment (sduskis) [inline src/row.js]: I'd revert this one, for now.  This changes the way rows are read.  The other changes changes aspects of table and family administration.

Once we're ready for this change, this would be a backwards incompatible change for row reads, which is a bit more tricky to convey to users.

File: src/table.js [hunk 2]
```diff
@@ -699,18 +699,18 @@ class Table {
...
-   * @param {string} name The family name.
+   * @param {string} id The family unique identifier.
    * @returns {Family}
    *
    * @example
    * const family = table.family('my-family');
    */
-  family(name) {
-    if (!name) {
+  family(id) {
```
Comment (sduskis) [inline src/table.js]: Please also change any uses of `family()` to only get the id instead of the full name.

File: system-test/bigtable.js [hunk 3]
```diff
@@ -388,18 +388,19 @@ describe('Bigtable', function() {
         assert.ifError(err);
         assert.strictEqual(families.length, 3);
         assert(families[0] instanceof Family);
-        assert.strictEqual(families[0].name, FAMILY.name);
+        assert.strictEqual(families[2].id, FAMILY.id);
```
Comment (sduskis) [inline system-test/bigtable.js]: Why was there a change from `families[0]` to `families[2]`?

File: test/row.js [hunk 4]
```diff
@@ -447,7 +447,7 @@ describe('Bigtable/Row', function() {
 
     var families = [
       {
-        name: 'test-family',
+        id: 'test-family',
```
Comment (sduskis) [inline test/row.js]: This should be reverted so that we can do a separate PR for families in ReadRows

File: system-test/bigtable.js (hunk 3 above)
Comment (vijay-qlogic) [reply system-test/bigtable.js]: because 2 families("follow", "traits") are already created above(ln. no. 61) in before section. And 3rd family("presidents") is created here, so index should be 2.
